pkgs/applications/networking/whoogle-search/default.nix

```diff
@@ -0,0 +1,76 @@
+{ lib
+, python3
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "whoogle-search";
+  version = "0.7.2";
+
+  src = python3.pkgs.fetchPypi {
+    inherit pname version;
+    hash = "sha256-RNmHo0SUHrYGNFgKBncpszNmjW0imqxhpnH+LS/oEK0=";
+  };
+
+  postInstall = ''
+    mv $out/${python3.sitePackages}/app/static $out/${python3.sitePackages}/app/static_runtime
+
+    # FIXME: Is it possible to specify dataDir from service configuration here?
+    ln -s /var/lib/whoogle-search/static $out/${python3.sitePackages}/app/static
+  '';
+
+  postPatch = ''
+    # Remove the version pinning
+    sed -i 's/[><=]=[0-9.]*//' requirements.txt
+
+    # Remove dev packages and packages installed by dependencies
+    cat <<EOF | sed --in-place --file - requirements.txt
+/pytest/d
+/pycodestyle/d
+/pluggy/d
+/packaging/d
+/py/d
+/more-itertools/d
+/pysocks/d
+/attrs/d
+/cachelib/d
+/certifi/d
+/cffi/d
+/chardet/d
+/click/d
+/idna/d
+/itsdangerous/d
+/jinja2/d
+/markupsafe/d
+/pycparser/d
+/pyopenssl/d
+/pyparsing/d
+/soupsieve/d
+/urllib3/d
+/wcwidth/d
+/werkzeug/d
+EOF
+  '';
+
+  propagatedBuildInputs = with python3.pkgs; [
+    beautifulsoup4
+    cssutils
+    cryptography
+    defusedxml
+    flask
+    flask-session
+    python-dotenv
+    requests
+    stem
+    waitress
+  ];
+
+  # Disable tests because they require network connection
+  doCheck = false;
+
```

Comment:
You can still add the NixOS test here (see https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking)

Reply:
Hello @FliegendeWurst ,
I am still not sure what do you exactly mean here.

Should I keep the `doCheck = false` and just add `passthru.tests = nixosTests.whoogle-search;` ? I already tried that (https://github.com/seberm/nixpkgs/commit/2d32b7b6786ce281d5c39747cb156d00a47340d1), but it seems the passthru option is not effective at all.